### PR TITLE
[editor] feat(initial intent): works for all plugins except file upload plugins

### PR DESCRIPTION
**Pull request by maimonav**

### CODE REVIEW

File: packages/common/web/src/Base/createBaseInsertPluginButton.jsx [hunk 1]
```diff
@@ -18,8 +19,21 @@ export default ({ blockType, button, helpers, pubsub, settings, t, isMobile }) =
       this.state = {};
       const { buttonStyles } = props.theme || {};
       this.styles = mergeStyles({ styles, theme: buttonStyles });
+      this.myRef = React.createRef();
```
Comment (bchmn) [inline packages/common/web/src/Base/createBaseInsertPluginButton.jsx]:
```suggestion
      this.buttonRef = React.createRef();
```

File: packages/editor/web/src/RichContentEditor/RichContentEditor.jsx [hunk 2]
```diff
@@ -47,6 +47,15 @@ class RichContentEditor extends Component {
     this.initContext();
     this.initPlugins();
   }
+  componentDidMount() {
+    this.resetInitialIntent();
+  }
+
+  resetInitialIntent = () => {
```
Comment (bchmn) [inline packages/editor/web/src/RichContentEditor/RichContentEditor.jsx]: why is this needed?

Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/RichContentEditor.jsx]: It is needed because initial intent as i understand, needs to happen only one time.
